workflow/validate/validate.go

```diff
@@ -574,6 +574,10 @@ func (tctx *templateValidationCtx) validateTemplateHolder(ctx context.Context, t
 	tmplCtx, resolvedTmpl, _, err := tmplCtx.ResolveTemplate(ctx, tmplHolder)
 	if err != nil {
 		if argoerr, ok := err.(errors.ArgoError); ok && argoerr.Code() == errors.CodeNotFound {
+			if tmplRef != nil && strings.Contains(tmplRef.Template, "placeholder") {
```

Comment:
Can we make it less likely to mismatch on non-placeholder templates with
```suggestion
			if tmplRef != nil && strings.HasPrefix(tmplRef.Template, "placeholder") {
```